lib/Chat/Notifier.php

```diff
@@ -53,13 +53,17 @@ public function __construct(
 	 * Not every user mentioned in the message is notified, but only those that
 	 * are able to participate in the room.
 	 *
+	 * @param Room $chat
+	 * @param IComment $comment
 	 * @param array[] $alreadyNotifiedUsers
 	 * @psalm-param array<int, array{id: string, type: string, reason: string, sourceId?: string, attendee?: Attendee}> $alreadyNotifiedUsers
+	 * @param bool $silent
+	 * @param Participant $participant
 	 * @return string[] Users that were mentioned
 	 * @psalm-return array<int, array{id: string, type: string, reason: string, sourceId?: string, attendee?: Attendee}>
 	 */
-	public function notifyMentionedUsers(Room $chat, IComment $comment, array $alreadyNotifiedUsers, bool $silent): array {
-		$usersToNotify = $this->getUsersToNotify($chat, $comment, $alreadyNotifiedUsers);
+	public function notifyMentionedUsers(Room $chat, IComment $comment, array $alreadyNotifiedUsers, bool $silent, Participant $participant): array {
```

Comment:
`$participant` can be null in the place calling this function. so you also need to allow null here and in the methods below and in case the permissions says "moderator" bots should not be able to mention?

Reply:
```suggestion
	public function notifyMentionedUsers(Room $chat, IComment $comment, array $alreadyNotifiedUsers, bool $silent, ?Participant $participant): array {
```